### PR TITLE
nixos/stage1: shellcheck fixes

### PR DESCRIPTION
**Pull request by Artturin**
###### Things done

@l0b0 

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Artturin** [comment]
the stage-1 script is run with ash instead of bash, however in shellcheck [ash is a alias to dash](https://github.com/koalaman/shellcheck/blob/23cddb037e93c2c1d3ae7c6bb4e159b9840e477b/src/ShellCheck/Data.hs#L134) and ash supports more features from bash than dash https://github.com/koalaman/shellcheck/issues/853

File: nixos/modules/system/boot/stage-1-init.sh [hunk 1]
```diff
@@ -152,22 +155,29 @@ exec > /tmp/stage-1-init.log.fifo 2>&1
 # Process the kernel command line.
 export stage2Init=/init
 for o in $(cat /proc/cmdline); do
+    # fixing SC2046/SC2086 (quote vars) breaks with
```
Comment (l0b0) [inline nixos/modules/system/boot/stage-1-init.sh:158]: You can treat the command line as an array by using the `mapfile` (alias `readarray`) builtin:

```
$ diff --ignore-blank-lines <(cat /proc/cmdline) <(mapfile args < /proc/cmdline && echo "${args[@]}")
$ echo $?
0
```

File: nixos/modules/system/boot/stage-1-init.sh [hunk 2]
```diff
@@ -152,22 +155,29 @@ exec > /tmp/stage-1-init.log.fifo 2>&1
...
+    # switch_root: can't execute '': No such file or directory
     case $o in
         console=*)
+            # shellcheck disable=SC2046,SC2086,SC1097
             set -- $(IFS==; echo $o)
```
Comment (l0b0) [inline nixos/modules/system/boot/stage-1-init.sh:163]: You can achieve the same with quoted variables by using `tr`:

```
$ o=bar=baz
$ echo $(IFS==; echo $o)
bar baz
$ tr = ' ' <<< "$o"
bar baz
```

File: nixos/modules/system/boot/stage-1-init.sh [hunk 3]
```diff
@@ -228,6 +240,7 @@ mkdir -p /lib
 ln -s @modulesClosure@/lib/modules /lib/modules
 ln -s @modulesClosure@/lib/firmware /lib/firmware
 echo @extraUtils@/bin/modprobe > /proc/sys/kernel/modprobe
+# shellcheck disable=SC2043
```
Comment (l0b0) [inline nixos/modules/system/boot/stage-1-init.sh:243]: This disables the rule for [the rest of the file, not just the next line](https://github.com/koalaman/shellcheck/issues/960). You might want to re-enable it after the loop or pull the loop into a function to disable only this line.

**Pull request closed without merging**